### DISCUSSION TITLE
Add graph mode for mapping relationships

### DISCUSSION THREAD

**Discussion by reesericci** [twentyhq/twenty#8332] [Ideas]
## Scope & Context

I'm looking to power map relationships between people, and I would love to be able to do this directly in Twenty. Something like a vector graphing mapmaker that could even just feed on a many-to-many field. 

## Current behavior

Mapping through tables/kanban only, no graph view

## Expected behavior

A diagramming tool that allows me to create custom maps between objects and maybe some sort of tooling for people inside a specific view.

**Example:** 

Vertedge is a simpler example, not feeding on a CRM: https://xarkenz.github.io/vertedge/

Here's some pictures of a power map:

![image](https://github.com/user-attachments/assets/45aadccc-d82d-4b59-a791-015a705e69ba)

![image](https://github.com/user-attachments/assets/3f86e512-0c1c-48fa-b0a8-b352fd009fcb)

## Technical inputs

Third-party contributor, don't know how to fill this.

**Comment by Bonapara** [comment]
What do you think of our data model graph?

`*/settings/objects/overview`

![CleanShot 2024-11-04 at 18 28 41](https://github.com/user-attachments/assets/dbcf5fd5-b645-4cbb-9225-c47a0d7b69c5)

**Comment by BOHEUS** [comment]
@Bonapara I think it's about relation between specific records 

@reesericci Many-to-many relation will be possible in the future, right now it's possible to have one-to-many or many-to-one relations between records/objects

**Comment by BOHEUS** [comment]
Even if it's something worth implementing, many-to-many relation, editable graph view and more queries in ORM must be implemented

**Comment by reesericci** [comment]
> What do you think of our data model graph?
> 
> `*/settings/objects/overview`
> 
> ![CleanShot 2024-11-04 at 18 28 41](https://private-user-images.githubusercontent.com/19412894/382856418-dbcf5fd5-b645-4cbb-9225-c47a0d7b69c5.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzA3NTE0ODIsIm5iZiI6MTczMDc1MTE4MiwicGF0aCI6Ii8xOTQxMjg5NC8zODI4NTY0MTgtZGJjZjVmZDUtYjY0NS00Y2JiLTkyMjUtYzQ3YTBkN2I2OWM1LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMDQlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTA0VDIwMTMwMlomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJjYWVjZmVmMzBiNmQxOGZlNjY5OWNhODQxNmVhMzkyNjc0MjVmMjBjNjdjNjEyODU1Y2FhNTUxZDllMzk2MzUmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.qUU_tox2dtp6d4OOStGuGpmcbJcOXD4yEl4rD1rEgi4)

I'm specifically looking for something that graphs individual records, not data types

**Comment by BOHEUS** [comment]
Related discussion https://github.com/twentyhq/twenty/discussions/11911
